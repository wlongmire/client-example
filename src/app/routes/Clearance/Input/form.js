import statesArray from './states';

const formData = {
	"questionSetId": "Clearance",
	"name": "Clearance",

    "questions": [
	  {
	    "questionId": "1",    
	    "text": "Who is the First Named Insured?",
	    "name": "primaryInsuredName",
	    "inputType": "text",
	    "inputFormat": "text",
	    "tooltiptext": "This entity must be named as the Owner in the contract receiving hold harmless, indemnification and additional insured status from the hired General Contractor",
	    "required": true
	   },
	    
	  
	  {
	    "questionId": "2",
	    "name": "primaryAddressLabel",
	    "text": "What is the address of the Project?",
	    "inputFormat": "label",
	    "attributes":{
	    	"controlGroup":"projectAddress"
	    }
	    
	  },

	  {
	    "questionId": "2a",
	    "name": "projectAddress",
	    "inputFormat": 	"text",
	    "inputType": 		"text",
	    "tooltiptext": "Please provide as descriptive of a street address as possible.",           
	    "required": true,
	    "placeholder": "Address",
	    "attributes":{
	    	"controlGroup":"projectAddress"	
	    }
	  },

	  {
	    "questionId": "2b",
	    "name": "projectCity",
	    "inputFormat": 	"text",
	    "inputType": 		"text",
	    "tooltiptext": "Please provide as descriptive of a street address as possible.",           
	    "required": true,
	    "placeholder": "City",
	    "attributes":{
	    	"controlGroup":"projectAddress"	
	    }
	  },

	  {
	    "questionId": "2c",
	    "name": "projectState",
	    "inputType": "dropdown-single",
	    "tooltiptext": "Please provide as descriptive of a street address as possible.",
	    "required": true,
	    "placeholder": "State",
	    "attributes":{	
	    	"options": statesArray,
	    	"controlGroup":"projectAddress"
	    }
	    
	  },

	  {
	    "questionId": "2d",
	    "name": "projectZipcode",
	    "inputType": 		"text",
	    "inputFormat": 	"text",
	    "tooltiptext": "Please provide as descriptive of a street address as possible.",
	    "required": true,
	    "placeholder": "Zip Code",
	    "attributes":{
	    	"controlGroup":"projectAddress"
	    }
	  }
	],

	"supplementalQuestions": []
}

export default formData;