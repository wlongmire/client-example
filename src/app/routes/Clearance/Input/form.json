{
	"questionSetId": "Clearance",
	"name": "Clearance",

    "questions": [
	  {
	    "questionId": "1",    
	    "text": "Who is the First Named Insured?",
	    "name": "nameInsuredName",
	    "inputType": "text",
	    "inputFormat": "text",
	    "tooltiptext": "This entity must be named as the Owner in the contract receiving hold harmless, indemnification and additional insured status from the hired General Contractor",
	    "required": true
	   },
	    
	  
	  {
	    "questionId": "2",
	    "name": "nameInsuredAddressLabel",
	    "text": "What is the address of the Named Insured?",
	    "inputFormat": "label",

	    "attributes":{
	    	"controlGroup":"address"
	    }
	    
	  },

	  {
	    "questionId": "2a",
	    "name": "nameInsuredAddress",
	    "inputFormat": 	"text",
	    "inputType": 		"text",
	    "tooltiptext": "Please provide as descriptive of a street address as possible.",           
	    "required": true,
	    "placeholder": "Address",
	    "attributes":{
	    	"controlGroup":"address"	
	    }
	  },

	  {
	    "questionId": "2b",
	    "name": "nameInsuredCity",
	    "inputFormat": 	"text",
	    "inputType": 		"text",
	    "tooltiptext": "Please provide as descriptive of a street address as possible.",           
	    "required": true,
	    "placeholder": "City",
	    "attributes":{
	    	"controlGroup":"address"	
	    }
	  },

	  {
	    "questionId": "2c",
	    "name": "namedInsuredState",
	    "inputType": "dropdown-single",
	    "tooltiptext": "Please provide as descriptive of a street address as possible.",
	    "required": true,
	    "placeholder": "State",
	    "attributes":{	
	    	"options":[
					{
	          "optionId": "0",
	          "text": "State",
	          "value": ""
	        },
	    		{
	          "optionId": "1",
	          "text": "Pennsylvania",
	          "value": "PA"
	        },
	        {
	          "optionId": "2",
	          "text": "New York",
	          "value": "NY"
	        }
	    	],
	    	"controlGroup":"address"
	    }
	    
	  },

	  {
	    "questionId": "2d",
	    "name": "nameInsuredZipcode",
	    "inputType": 		"text",
	    "inputFormat": 	"text",
	    "tooltiptext": "Please provide as descriptive of a street address as possible.",
	    "required": true,
	    "placeholder": "Zip Code",
	    "attributes":{
	    	"controlGroup":"address"
	    }
	  }
	],

	"supplementalQuestions": []
}