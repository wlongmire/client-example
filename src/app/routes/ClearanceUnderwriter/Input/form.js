import statesArray from './states';

const formData = {
	"questionSetId": "Clearance",
	"name": "Clearance",

    "questions": [
	  {
	    "questionId": "1",    
	    "text": "Who is the First Named Insured?",
	    "name": "primaryInsuredName",
	    "inputType": "text",
	    "inputFormat": "text",
	    "tooltiptext": "This entity must be named as the Owner in the contract receiving hold harmless, indemnification and additional insured status from the hired General Contractor",
	    "required": true
	   },
	    
	  {
	    "questionId": "2",
	    "name": "primaryAddressLabel",
	    "text": "What is the First Named Insured's Address?",
	    "inputFormat": "label",
	    "attributes":{
	    	"controlGroup":"insuredAddress"
	    }
	    
	  },

	  {
	    "questionId": "2a",
	    "name": "primaryInsuredAddress",
	    "inputFormat": 	"text",
	    "inputType": 		"text",
	    "tooltiptext": "Please provide as descriptive of a street address as possible.",           
	    "required": true,
	    "placeholder": "Address",
	    "attributes":{
	    	"controlGroup":"insuredAddress"	
	    }
	  },

	  {
	    "questionId": "2b",
	    "name": "primaryInsuredCity",
	    "inputFormat": 	"text",
	    "inputType": 		"text",
	    "tooltiptext": "Please provide as descriptive of a street address as possible.",           
	    "required": true,
	    "placeholder": "City",
	    "attributes":{
	    	"controlGroup":"insuredAddress"	
	    }
	  },

	  {
	    "questionId": "2c",
	    "name": "primaryInsuredState",
	    "inputType": "dropdown-single",
	    "tooltiptext": "Please provide as descriptive of a street address as possible.",
	    "required": true,
	    "placeholder": "State",
	    "attributes":{	
	    	"options": statesArray,
	    	"controlGroup":"insuredAddress"
	    }
	    
	  },

	  {
	    "questionId": "2d",
	    "name": "primaryInsuredZipcode",
	    "inputType": 		"text",
	    "inputFormat": 	"text",
	    "tooltiptext": "Please provide as descriptive of a street address as possible.",
	    "required": true,
	    "placeholder": "Zip Code",
	    "attributes":{
	    	"controlGroup":"insuredAddress"
	    }
	  },

	  {
	    "questionId": "3",
	    "name": "primaryAddressLabel",
	    "text": "What is the Project Address?",
	    "inputFormat": "label",
	    "attributes":{
	    	"controlGroup":"projectAddress"
	    }
	    
	  },

	  {
	    "questionId": "3a",
	    "name": "projectAddress",
	    "inputFormat": 	"text",
	    "inputType": 		"text",
	    "tooltiptext": "Please provide as descriptive of a street address as possible.",           
	    "required": true,
	    "placeholder": "Address",
	    "attributes":{
	    	"controlGroup":"projectAddress"	
	    }
	  },

	  {
	    "questionId": "3b",
	    "name": "projectCity",
	    "inputFormat": 	"text",
	    "inputType": 		"text",
	    "tooltiptext": "Please provide as descriptive of a street address as possible.",           
	    "required": true,
	    "placeholder": "City",
	    "attributes":{
	    	"controlGroup":"projectAddress"	
	    }
	  },

	  {
	    "questionId": "3c",
	    "name": "projectState",
	    "inputType": "dropdown-single",
	    "tooltiptext": "Please provide as descriptive of a street address as possible.",
	    "required": true,
	    "placeholder": "State",
	    "attributes":{	
	    	"options": statesArray,
	    	"controlGroup":"projectAddress"
	    }
	    
	  },

	  {
	    "questionId": "3d",
	    "name": "projectZipcode",
	    "inputType": 		"text",
	    "inputFormat": 	"text",
	    "tooltiptext": "Please provide as descriptive of a street address as possible.",
	    "required": true,
	    "placeholder": "Zip Code",
	    "attributes":{
	    	"controlGroup":"projectAddress"
	    }
	  }
	],

	"supplementalQuestions": []
}

export default formData;