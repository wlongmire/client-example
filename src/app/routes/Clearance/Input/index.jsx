import React, { Component } from 'react'
import DialogBox from 'app/components/shared/DialogBox'
import { connect } from 'react-redux'
import { ButtonGroup, Button } from 'react-bootstrap'
import FormBuilder from 'app/components/shared/FormBuilder'
import form from './form.js'

const STATUS = {
  INPUT: "INPUT",
  LOADING: "LOADING",
  ERROR: "ERROR",
  RESULT: "RESULT"
}

export default class Input extends Component {
    constructor(props) {
      super(props)
      this.state = {
        requiredFields: [],
        validationModal: false,
        status: STATUS.INPUT
      }

      this.handleSubmit = this.handleSubmit.bind(this)
      this.handleValidationOk = this.handleValidationOk.bind(this)
    }

  handleSubmit(values, controlGroups, requiredFields) {
    if (requiredFields.length > 0) {
      this.setState({
        ...this.state,
        requiredFields,
        validationModal: true
      })
    } else {
      console.log('VALUES', values)
      this.props.handleSubmit(values)
    }
  }

    handleValidationOk() {
      this.setState({
        ...this.state,
        validationModal: false
      })
    }

    render() {
      const requiredList = () => {
        return this.state.requiredFields.map((r, idx) => {
          const fieldText = (r.questionId == '2c') ? 'State' : r.text

          return (
              <li key={idx} className="remainingField">{(fieldText ? fieldText : r.placeholder)}</li>
            )
        })
      }

      return (
        <div>
            <h3>First Let's Check for Clearance.</h3>
            <h4>Enter the following information to clear against previous submissions.</h4>

            <FormBuilder
                data={form}
                initialValues={this.props.input}
                submitTitle="Check For Clearance"
                handleSubmit={this.handleSubmit}
            />


          <DialogBox
            custom_class="confirmationDialog"
            title="Please fill out all required fields."
            show={this.state.validationModal}
          >
            <div> 
              <h4>Here are your remaining questions:</h4>
              <ul className="section">
                { requiredList() }
              </ul>

              <h4>
                Note: All required fields are <span className="required">underlined in red.</span>
              </h4>
              <br />
              <ButtonGroup>
                <Button
                  className="btn secondary"
                  onClick={this.handleValidationOk}
                >
                  Return to the Form
                </Button>
              </ButtonGroup>
            </div>

          </DialogBox>

        </div>)
    }
}
