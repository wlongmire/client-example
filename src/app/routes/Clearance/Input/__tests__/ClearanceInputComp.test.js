/* eslint-env node, jest */

import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Input from '../index'

const initialState = {
  user: {
    _id: '5807af4bbd9f641100e26138',
    username: 'betamarshuser',
    firstName: 'Marsh',
    lastName: 'Beta',
    role: 'user',
    _brokerId: '5807af2edcba0f490c718471',
    accountPending: false
  },
  submissionData: [
    {
      _id: '59413679bafa390001ced695',
      primaryInsuredName: 'Andriy Kulak',
      totalCost: '$23,213,234',
      quotedPremium: '$21,612',
      totalPremium: '$22,693',
      type: 'ocp',
      updatedAt: '2017-06-14T13:13:29.776Z',
      dateCreated: '06-14-17 09:13am',
      dateUpdated: '06-14-17 09:13am',
      quoteStatus: 'Yes'
    }
  ]
}

// Snapshot for Home Header Component
describe('>>> Clearance Component --- Snapshot', () => {
  it('+++capturing Snapshot with', () => {
    const renderedValue = shallow(
      <Input
        submissions={initialState.submissionData}
        handleSubmit={jest.fn()}
        getSubmissions={jest.fn()}
        user={initialState.submissionData}
      />)
    expect(toJson(renderedValue)).toMatchSnapshot()
  })
})
