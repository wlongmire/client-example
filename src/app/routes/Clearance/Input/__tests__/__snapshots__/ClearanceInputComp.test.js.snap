// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>> Clearance Component --- Snapshot +++capturing Snapshot with 1`] = `
<div>
  <h3>
    First Let's Check for Clearance.
  </h3>
  <h4>
    Enter the following information to clear against previous submissions.
  </h4>
  <FormBuilder
    data={
      Object {
        "name": "Clearance",
        "questionSetId": "Clearance",
        "questions": Array [
          Object {
            "inputFormat": "text",
            "inputType": "text",
            "name": "primaryInsuredName",
            "questionId": "1",
            "required": true,
            "text": "Who is the First Named Insured?",
            "tooltiptext": "This entity must be named as the Owner in the contract receiving hold harmless, indemnification and additional insured status from the hired General Contractor",
          },
          Object {
            "attributes": Object {
              "controlGroup": "insuredAddress",
            },
            "inputFormat": "label",
            "name": "primaryAddressLabel",
            "questionId": "2",
            "text": "What is the First Named Insured's Address?",
          },
          Object {
            "attributes": Object {
              "controlGroup": "insuredAddress",
            },
            "inputFormat": "text",
            "inputType": "text",
            "name": "primaryInsuredAddress",
            "placeholder": "Address",
            "questionId": "2a",
            "required": true,
            "tooltiptext": "Please provide as descriptive of a street address as possible.",
          },
          Object {
            "attributes": Object {
              "controlGroup": "insuredAddress",
            },
            "inputFormat": "text",
            "inputType": "text",
            "name": "primaryInsuredCity",
            "placeholder": "City",
            "questionId": "2b",
            "required": true,
            "tooltiptext": "Please provide as descriptive of a street address as possible.",
          },
          Object {
            "attributes": Object {
              "controlGroup": "insuredAddress",
              "options": Array [
                Object {
                  "optionId": "1",
                  "text": "State",
                  "value": "",
                },
                Object {
                  "optionId": "2",
                  "text": "Alabama",
                  "value": "Alabama",
                },
                Object {
                  "optionId": "3",
                  "text": "Alaska",
                  "value": "Alaska",
                },
                Object {
                  "optionId": "4",
                  "text": "Arizona",
                  "value": "Arizona",
                },
                Object {
                  "optionId": "5",
                  "text": "Arkansas",
                  "value": "Arizona",
                },
                Object {
                  "optionId": "6",
                  "text": "California",
                  "value": "California",
                },
                Object {
                  "optionId": "7",
                  "text": "Colorado",
                  "value": "Colorado",
                },
                Object {
                  "optionId": "8",
                  "text": "Connecticut",
                  "value": "Connecticut",
                },
                Object {
                  "optionId": "9",
                  "text": "Delaware",
                  "value": "Delaware",
                },
                Object {
                  "optionId": "10",
                  "text": "District of Columbia",
                  "value": "District of Columbia",
                },
                Object {
                  "optionId": "11",
                  "text": "Florida",
                  "value": "Florida",
                },
                Object {
                  "optionId": "12",
                  "text": "Georgia",
                  "value": "Georgia",
                },
                Object {
                  "optionId": "13",
                  "text": "Hawaii",
                  "value": "Hawaii",
                },
                Object {
                  "optionId": "14",
                  "text": "Idaho",
                  "value": "Idaho",
                },
                Object {
                  "optionId": "15",
                  "text": "Illinois",
                  "value": "Illinois",
                },
                Object {
                  "optionId": "16",
                  "text": "Indiana",
                  "value": "Indiana",
                },
                Object {
                  "optionId": "17",
                  "text": "Iowa",
                  "value": "Iowa",
                },
                Object {
                  "optionId": "18",
                  "text": "Kansas",
                  "value": "Kansas",
                },
                Object {
                  "optionId": "19",
                  "text": "Kentucky",
                  "value": "Kentucky",
                },
                Object {
                  "optionId": "20",
                  "text": "Louisiana",
                  "value": "Louisiana",
                },
                Object {
                  "optionId": "21",
                  "text": "Maine",
                  "value": "Maine",
                },
                Object {
                  "optionId": "22",
                  "text": "Maryland",
                  "value": "Maryland",
                },
                Object {
                  "optionId": "23",
                  "text": "Massachusetts",
                  "value": "Massachusetts",
                },
                Object {
                  "optionId": "24",
                  "text": "Michigan",
                  "value": "Michigan",
                },
                Object {
                  "optionId": "25",
                  "text": "Minnesota",
                  "value": "Minnesota",
                },
                Object {
                  "optionId": "26",
                  "text": "Mississippi",
                  "value": "Mississippi",
                },
                Object {
                  "optionId": "27",
                  "text": "Missouri",
                  "value": "Missouri",
                },
                Object {
                  "optionId": "28",
                  "text": "Montana",
                  "value": "Montana",
                },
                Object {
                  "optionId": "29",
                  "text": "Nebraska",
                  "value": "Nebraska",
                },
                Object {
                  "optionId": "30",
                  "text": "Nevada",
                  "value": "Nevada",
                },
                Object {
                  "optionId": "31",
                  "text": "New Hampshire",
                  "value": "New Hampshire",
                },
                Object {
                  "optionId": "32",
                  "text": "New Jersey",
                  "value": "New Jersey",
                },
                Object {
                  "optionId": "33",
                  "text": "New Mexico",
                  "value": "New Mexico",
                },
                Object {
                  "optionId": "34",
                  "text": "New York",
                  "value": "New York",
                },
                Object {
                  "optionId": "35",
                  "text": "North Carolina",
                  "value": "North Carolina",
                },
                Object {
                  "optionId": "36",
                  "text": "North Dakota",
                  "value": "North Dakota",
                },
                Object {
                  "optionId": "37",
                  "text": "Ohio",
                  "value": "Ohio",
                },
                Object {
                  "optionId": "38",
                  "text": "Oklahoma",
                  "value": "Oklahoma",
                },
                Object {
                  "optionId": "39",
                  "text": "Oregon",
                  "value": "Oregon",
                },
                Object {
                  "optionId": "40",
                  "text": "Pennsylvania",
                  "value": "Pennsylvania",
                },
                Object {
                  "optionId": "41",
                  "text": "Rhode Island",
                  "value": "Rhode Island",
                },
                Object {
                  "optionId": "42",
                  "text": "South Carolina",
                  "value": "South Carolina",
                },
                Object {
                  "optionId": "43",
                  "text": "South Dakota",
                  "value": "South Dakota",
                },
                Object {
                  "optionId": "44",
                  "text": "Tennessee",
                  "value": "Tennessee",
                },
                Object {
                  "optionId": "45",
                  "text": "Texas",
                  "value": "Texas",
                },
                Object {
                  "optionId": "46",
                  "text": "Utah",
                  "value": "Utah",
                },
                Object {
                  "optionId": "47",
                  "text": "Vermont",
                  "value": "Vermont",
                },
                Object {
                  "optionId": "48",
                  "text": "Virginia",
                  "value": "Virginia",
                },
                Object {
                  "optionId": "49",
                  "text": "Washington",
                  "value": "Washington",
                },
                Object {
                  "optionId": "50",
                  "text": "West Virginia",
                  "value": "West Virginia",
                },
                Object {
                  "optionId": "51",
                  "text": "Wisconsin",
                  "value": "Wisconsin",
                },
                Object {
                  "optionId": "52",
                  "text": "Wyoming",
                  "value": "Wyoming",
                },
              ],
            },
            "inputType": "dropdown-single",
            "name": "primaryInsuredState",
            "placeholder": "State",
            "questionId": "2c",
            "required": true,
            "tooltiptext": "Please provide as descriptive of a street address as possible.",
          },
          Object {
            "attributes": Object {
              "controlGroup": "insuredAddress",
            },
            "inputFormat": "text",
            "inputType": "text",
            "name": "primaryInsuredZipcode",
            "placeholder": "Zip Code",
            "questionId": "2d",
            "required": true,
            "tooltiptext": "Please provide as descriptive of a street address as possible.",
          },
          Object {
            "attributes": Object {
              "controlGroup": "projectAddress",
            },
            "inputFormat": "label",
            "name": "primaryAddressLabel",
            "questionId": "3",
            "text": "What is the Project Address?",
          },
          Object {
            "attributes": Object {
              "controlGroup": "projectAddress",
            },
            "inputFormat": "text",
            "inputType": "text",
            "name": "projectAddress",
            "placeholder": "Address",
            "questionId": "3a",
            "required": true,
            "tooltiptext": "Please provide as descriptive of a street address as possible.",
          },
          Object {
            "attributes": Object {
              "controlGroup": "projectAddress",
            },
            "inputFormat": "text",
            "inputType": "text",
            "name": "projectCity",
            "placeholder": "City",
            "questionId": "3b",
            "required": true,
            "tooltiptext": "Please provide as descriptive of a street address as possible.",
          },
          Object {
            "attributes": Object {
              "controlGroup": "projectAddress",
              "options": Array [
                Object {
                  "optionId": "1",
                  "text": "State",
                  "value": "",
                },
                Object {
                  "optionId": "2",
                  "text": "Alabama",
                  "value": "Alabama",
                },
                Object {
                  "optionId": "3",
                  "text": "Alaska",
                  "value": "Alaska",
                },
                Object {
                  "optionId": "4",
                  "text": "Arizona",
                  "value": "Arizona",
                },
                Object {
                  "optionId": "5",
                  "text": "Arkansas",
                  "value": "Arizona",
                },
                Object {
                  "optionId": "6",
                  "text": "California",
                  "value": "California",
                },
                Object {
                  "optionId": "7",
                  "text": "Colorado",
                  "value": "Colorado",
                },
                Object {
                  "optionId": "8",
                  "text": "Connecticut",
                  "value": "Connecticut",
                },
                Object {
                  "optionId": "9",
                  "text": "Delaware",
                  "value": "Delaware",
                },
                Object {
                  "optionId": "10",
                  "text": "District of Columbia",
                  "value": "District of Columbia",
                },
                Object {
                  "optionId": "11",
                  "text": "Florida",
                  "value": "Florida",
                },
                Object {
                  "optionId": "12",
                  "text": "Georgia",
                  "value": "Georgia",
                },
                Object {
                  "optionId": "13",
                  "text": "Hawaii",
                  "value": "Hawaii",
                },
                Object {
                  "optionId": "14",
                  "text": "Idaho",
                  "value": "Idaho",
                },
                Object {
                  "optionId": "15",
                  "text": "Illinois",
                  "value": "Illinois",
                },
                Object {
                  "optionId": "16",
                  "text": "Indiana",
                  "value": "Indiana",
                },
                Object {
                  "optionId": "17",
                  "text": "Iowa",
                  "value": "Iowa",
                },
                Object {
                  "optionId": "18",
                  "text": "Kansas",
                  "value": "Kansas",
                },
                Object {
                  "optionId": "19",
                  "text": "Kentucky",
                  "value": "Kentucky",
                },
                Object {
                  "optionId": "20",
                  "text": "Louisiana",
                  "value": "Louisiana",
                },
                Object {
                  "optionId": "21",
                  "text": "Maine",
                  "value": "Maine",
                },
                Object {
                  "optionId": "22",
                  "text": "Maryland",
                  "value": "Maryland",
                },
                Object {
                  "optionId": "23",
                  "text": "Massachusetts",
                  "value": "Massachusetts",
                },
                Object {
                  "optionId": "24",
                  "text": "Michigan",
                  "value": "Michigan",
                },
                Object {
                  "optionId": "25",
                  "text": "Minnesota",
                  "value": "Minnesota",
                },
                Object {
                  "optionId": "26",
                  "text": "Mississippi",
                  "value": "Mississippi",
                },
                Object {
                  "optionId": "27",
                  "text": "Missouri",
                  "value": "Missouri",
                },
                Object {
                  "optionId": "28",
                  "text": "Montana",
                  "value": "Montana",
                },
                Object {
                  "optionId": "29",
                  "text": "Nebraska",
                  "value": "Nebraska",
                },
                Object {
                  "optionId": "30",
                  "text": "Nevada",
                  "value": "Nevada",
                },
                Object {
                  "optionId": "31",
                  "text": "New Hampshire",
                  "value": "New Hampshire",
                },
                Object {
                  "optionId": "32",
                  "text": "New Jersey",
                  "value": "New Jersey",
                },
                Object {
                  "optionId": "33",
                  "text": "New Mexico",
                  "value": "New Mexico",
                },
                Object {
                  "optionId": "34",
                  "text": "New York",
                  "value": "New York",
                },
                Object {
                  "optionId": "35",
                  "text": "North Carolina",
                  "value": "North Carolina",
                },
                Object {
                  "optionId": "36",
                  "text": "North Dakota",
                  "value": "North Dakota",
                },
                Object {
                  "optionId": "37",
                  "text": "Ohio",
                  "value": "Ohio",
                },
                Object {
                  "optionId": "38",
                  "text": "Oklahoma",
                  "value": "Oklahoma",
                },
                Object {
                  "optionId": "39",
                  "text": "Oregon",
                  "value": "Oregon",
                },
                Object {
                  "optionId": "40",
                  "text": "Pennsylvania",
                  "value": "Pennsylvania",
                },
                Object {
                  "optionId": "41",
                  "text": "Rhode Island",
                  "value": "Rhode Island",
                },
                Object {
                  "optionId": "42",
                  "text": "South Carolina",
                  "value": "South Carolina",
                },
                Object {
                  "optionId": "43",
                  "text": "South Dakota",
                  "value": "South Dakota",
                },
                Object {
                  "optionId": "44",
                  "text": "Tennessee",
                  "value": "Tennessee",
                },
                Object {
                  "optionId": "45",
                  "text": "Texas",
                  "value": "Texas",
                },
                Object {
                  "optionId": "46",
                  "text": "Utah",
                  "value": "Utah",
                },
                Object {
                  "optionId": "47",
                  "text": "Vermont",
                  "value": "Vermont",
                },
                Object {
                  "optionId": "48",
                  "text": "Virginia",
                  "value": "Virginia",
                },
                Object {
                  "optionId": "49",
                  "text": "Washington",
                  "value": "Washington",
                },
                Object {
                  "optionId": "50",
                  "text": "West Virginia",
                  "value": "West Virginia",
                },
                Object {
                  "optionId": "51",
                  "text": "Wisconsin",
                  "value": "Wisconsin",
                },
                Object {
                  "optionId": "52",
                  "text": "Wyoming",
                  "value": "Wyoming",
                },
              ],
            },
            "inputType": "dropdown-single",
            "name": "projectState",
            "placeholder": "State",
            "questionId": "3c",
            "required": true,
            "tooltiptext": "Please provide as descriptive of a street address as possible.",
          },
          Object {
            "attributes": Object {
              "controlGroup": "projectAddress",
            },
            "inputFormat": "text",
            "inputType": "text",
            "name": "projectZipcode",
            "placeholder": "Zip Code",
            "questionId": "3d",
            "required": true,
            "tooltiptext": "Please provide as descriptive of a street address as possible.",
          },
        ],
        "supplementalQuestions": Array [],
      }
    }
    handleSubmit={[Function]}
    submitTitle="Check For Clearance"
  />
  <DialogBox
    custom_class="confirmationDialog"
    show={false}
    title="Please fill out all required fields."
  >
    <div>
      <h4>
        Here are your remaining questions:
      </h4>
      <ul
        className="section"
      />
      <h4>
        Note: All required fields are 
        <span
          className="required"
        >
          underlined in red.
        </span>
      </h4>
      <br />
      <ButtonGroup
        block={false}
        bsClass="btn-group"
        justified={false}
        vertical={false}
      >
        <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="default"
          className="btn secondary"
          disabled={false}
          onClick={[Function]}
        >
          Return to the Form
        </Button>
      </ButtonGroup>
    </div>
  </DialogBox>
</div>
`;
